gameplay/src/PlatformLinux.cpp

```diff
@@ -1333,6 +1334,13 @@ int Platform::enterMessagePump()
             _game->frame();
         }
 
+        if (!visible && __vsync) {
+        	// glXSwapBuffers may not block on vsync properly if the window is obscured/offdesktop --
+        	// if vsync enabled, do an artificial slowdown - 15ms is largely arbitrary
+        	// if the we have disabled vsync, we don't care, run full speed anyway.
+        	usleep(15000.0);
```

Comment:
By the way, I just checked under Ubuntu 14.04 with compiz window manager. In that case the compiz window manager doesn't send the UnmapNotify event at all when the app is in the background or minimized. Probably because it's still compositing those windows to show in the task switcher etc. So setting visible = false on UnmapNotify doesn't have any effect.